### PR TITLE
Tasks v2beta2

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**
Add synth.py and regen gapic.

### CODE REVIEW

File: google-cloud-tasks/synth.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+v1_library = gapic.ruby_library(
+    'tasks', 'v2beta2', artman_output_name='google-cloud-ruby/google-cloud-tasks',
+    config_path='artman_cloudtasks.yaml'
+)
+
+s.copy(v1_library)
+
+# https://github.com/googleapis/gapic-generator/issues/2080
+s.replace(
```
Comment (dazuma) [inline google-cloud-tasks/synth.py]: This appears to be not needed anymore.

File: google-cloud-tasks/synth.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+logging.basicConfig(level=logging.DEBUG)
+
+gapic = gcp.GAPICGenerator()
+
+v1_library = gapic.ruby_library(
+    'tasks', 'v2beta2', artman_output_name='google-cloud-ruby/google-cloud-tasks',
+    config_path='artman_cloudtasks.yaml'
+)
+
+s.copy(v1_library)
```
Comment (dazuma) [inline google-cloud-tasks/synth.py]: I think we should avoid replacing the Gemfile, gemspec, readme, etc., even for gapic-only. Just replace the code (in lib and test).

File: google-cloud-tasks/README.md [hunk 3]
```diff
@@ -11,7 +12,7 @@ steps:
 
 1. [Select or create a Cloud Platform project.](https://console.cloud.google.com/project)
 2. [Enable billing for your project.](https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project)
-3. [Enable the Cloud Tasks API.](https://console.cloud.google.com/apis/api/tasks)
+3. [Enable the Cloud Tasks API.](https://console.cloud.google.com/apis/api/cloudtasks)
```
Comment (blowmage) [inline google-cloud-tasks/README.md]: This URL doesn't work.

File: google-cloud-tasks/README.md [hunk 4]
```diff
@@ -28,41 +29,4 @@ $ gem install google-cloud-tasks
   to see the full list of Cloud APIs that we cover.
 
 [Client Library Documentation]: https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud-tasks/latest/google/cloud/tasks/v2beta2
-[Product Documentation]: https://cloud.google.com/tasks
-
-## Enabling Logging
```
Comment (blowmage) [inline google-cloud-tasks/README.md]: Why is this section removed?

File: google-cloud-tasks/lib/google/cloud/tasks/v2beta2/credentials.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2018 Google LLC
```
Comment (blowmage) [inline google-cloud-tasks/lib/google/cloud/tasks/v2beta2/credentials.rb]: Now that this file has been moved the old file `lib/google/cloud/tasks/credentials.rb` should be removed.

**Comment by blowmage** [comment]
The PR's first commit (42850e87924b38d2f5e0262f88a89cb07370edac) updated the README by fixing some links and added some missing sections, but a later commit (f962cbc95ef2bb4536a4ee667218a4e8bdf81bf8) removed those changes. Why? Now the links in the documentation (which look to be correct) are inconsistent with the links in the README (which looks to be incorrect).

**Comment by dazuma** [comment]
I'm trying to decide whether it makes sense to regenerate files like the readmes (and gemspec, etc.) because it would clobber any manual edits we may have made in the meantime. In the case of this specific PR, yes, we should reinstate those edits, especially because it includes fixes to links. But in the long term, I'm not sure if this should be done automatically. Sorry if I'm not up to date—has this been discussed previously?

File: google-cloud-tasks/README.md (hunk 4 above)
Comment (TheRoyalTnetennba) [reply google-cloud-tasks/README.md]: Was overly conservative during the generation. Re-added newest README.

**Pull request merged**